scheduler/src/managers/node_agent/metrics.rs

```diff
@@ -35,14 +40,32 @@ pub struct NodeAgent {
 
 impl NodeAgent {
     /// Create a new `NodeAgent`.
-    pub fn new() -> Self {
+    ///
+    /// # Arguments
+    ///
+    /// * `id` - The ID of the node agent.
+    /// * `address` - The address where the node agent is reachable at.
+    pub fn new(id: String, address: SocketAddr) -> Self {
         NodeAgent {
+            id,
+            address,
             last_heartbeat: Local::now(),
             memory: None,
             cpu: None,
         }
     }
 
+    /// Get the agent's id.
+    pub fn id(&self) -> &str {
+        &self.id
+    }
+
+    /// Get the agent's formatted server address.
+    /// This is the URL that should be used to contact its gRPC server.
+    pub fn address(&self) -> String {
+        "http://".to_string() + &*self.address.to_string()
+    }
```

Comment:
Please rename the method and use `format!` instead.
```suggestion
    pub fn grpc_url(&self) -> String {
        format!("http://{}", self.address.to_string())
    }
```

Reply:
Done! Sorry, that was yucky